payjoin-cli/src/app.rs

```diff
@@ -211,12 +227,18 @@ impl App {
             println!("Sent fallback transaction");
             match ctx.process_response(&mut response.into_reader()) {
                 Ok(Some(psbt)) => return Ok(psbt),
-                Ok(None) => std::thread::sleep(std::time::Duration::from_secs(5)),
-                Err(re) => {
-                    println!("{}", re);
-                    log::debug!("{:?}", re);
+                Ok(None) => {
+                    std::thread::sleep(std::time::Duration::from_secs(5));
+                    continue;
                 }
-            }
+                Err(error) => {
+                    if error.is_version_supported(&1) {
+                        return self.try_v1(req_ctx).await;
+                    };
+                    println!("{}", error);
+                    log::debug!("{:?}", error);
```

Comment:
returning early on line 236 here will skip printing the error if one is encountered. Rather, if an error is encountered there it should run on the same log statements. This will also require `try_v1` returns Result<Psbt, ResponseError> instead of an anyhow::Error too. Then the last line of try_v1 can just be the process function and does not need to be wrapped in Ok()

```rust
Err(mut error) => {
    if error.is_version_supported(&1) {
        match self.try_v1(req_ctx).await {
            Ok(res) => return Ok(res),
            Err(e) => error = e,
        }
    };
    println!("{}", error);
    log::debug!("{:?}", error);
}
```